Cabal/Distribution/PackageDescription/Parsec.hs

```diff
@@ -297,7 +300,18 @@ parseGenericPackageDescription' lexWarnings fs = do
                 _ -> do
                     parseFailure pos $ "Invalid source-repository kind " ++ show args
                     pure RepoHead
-            sr <- parseFields sourceRepoFieldDescrs warnUnrec (emptySourceRepo kind) fields
```

Comment:
also note that one won't need *fishy* `emptySourceRepo` anymore, and similar for other types!